### ISSUE TITLE
Build fails: Multiple applicable methods for sample_format()

### ISSUE THREAD

**Issue by killkrt** [RustAudio/sound_stream#38]
Hi,

It seems that there is a conflict between `portaudio::pa::Sample`and `sample::Sample`, they both define `sample_format()`method.
This is the output of `cargo build`:

```
Compiling sound_stream v0.3.7 (file:///Users/KillKRT/Documents/Development/rust/sound_stream)
src/event.rs:131:57: 131:72 error: multiple applicable methods in scope [E0034]
src/event.rs:131         let input_sample_format  = default_input_sample.sample_format();
                                                                         ^~~~~~~~~~~~~~~
src/event.rs:131:57: 131:72 note: candidate #1 is defined in the trait `portaudio::pa::Sample`
src/event.rs:131         let input_sample_format  = default_input_sample.sample_format();
                                                                         ^~~~~~~~~~~~~~~
src/event.rs:131:57: 131:72 note: candidate #2 is defined in the trait `sample::Sample`
src/event.rs:131         let input_sample_format  = default_input_sample.sample_format();
                                                                         ^~~~~~~~~~~~~~~
src/event.rs:132:58: 132:73 error: multiple applicable methods in scope [E0034]
src/event.rs:132         let output_sample_format = default_output_sample.sample_format();
                                                                          ^~~~~~~~~~~~~~~
src/event.rs:132:58: 132:73 note: candidate #1 is defined in the trait `portaudio::pa::Sample`
src/event.rs:132         let output_sample_format = default_output_sample.sample_format();
                                                                          ^~~~~~~~~~~~~~~
src/event.rs:132:58: 132:73 note: candidate #2 is defined in the trait `sample::Sample`
src/event.rs:132         let output_sample_format = default_output_sample.sample_format();
                                                                          ^~~~~~~~~~~~~~~
src/event.rs:168:39: 168:63 error: type `portaudio::pa::types::stream_flags::StreamFlags` does not implement any method in scope named `ClipOff`
src/event.rs:168                                       pa::StreamFlags::ClipOff,
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to 3 previous errors
Could not compile `sound_stream`.
```

Since the project has the travis "build passed" badge, I guess something has changed recently in [rust-portaudio](https://github.com/jeremyletang/rust-portaudio) (current version is `0.4.7`). I've tried to modify the dependency version of `portaudio` in `Cargo.toml`(and made a `cargo update`), but Cargo still compile version `0.4.7`.

System info:
OS: OS X 10.10.3
Rust: rustc 1.0.0-beta.2 (e9080ec39 2015-04-16) (built 2015-04-16)

**Comment by mitchmindtree** [comment]
Hey @killkrt, I'm currently reworking sound_stream's innards a little at the moment, I'll be sure to get around to this (and use UFCS for the sample_format method calls), I should have a fix up within 24hrs :)

**Comment by killkrt** [comment]
Thank you for your feedback! You're always very responsive!

**Comment by mitchmindtree** [comment]
@killkrt no problem! This has been closed by #39.

**Comment by killkrt** [comment]
Thanks!
